robottelo/host_helpers/satellite_mixins.py

```diff
@@ -141,8 +141,9 @@ def upload_manifest(self, org_id, manifest=None, interface='API', timeout=None):
         :returns: the manifest upload result
 
         """
-        if manifest.content is None:
-            manifest = clone()
+        if not isinstance(manifest, (bytes, io.BytesIO)):
+            if manifest.content is None:
+                manifest = clone()
```

Comment:
Can we instead check manifest has content atribute set using hasattr here? non-blocking
```suggestion
        if hasattr(manifest, 'content')
            manifest = clone()
```

Reply:
This suggestion would substitute cloned manifests for manifester manifests in some situations where doing so is not necessary. For example: https://github.com/SatelliteQE/robottelo/blob/7a5840a9b83fa9fb0902c7a2ff13ec51b2b93677/tests/foreman/cli/test_contentaccess.py#L191
In this test, the object passed as the `manifest` argument to `upload_manifest` will have a `content` attribute if manifester succeeded and would then get swapped out for a cloned manifest at this point even with a valid manifest from manifester.

Reply:
ah gotcha, my bad, suggested condition should be something like ` if hasattr(manifest, 'content') is False:`, so it can work when when manifest.content is passed as an argument to upload_manifest() and will know if manifest is passed which will have a content so won't clone it, right?